Sources/PubNub/KMP/Wrappers/KMPPresenceChange.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  PubNubPresenceChangeObjC.swift
+//
+//  Copyright (c) PubNub Inc.
+//  All rights reserved.
+//
+//  This source code is licensed under the license found in the
+//  LICENSE file in the root directory of this source tree.
+//
+// IMPORTANT NOTE FOR DEVELOPERS USING THIS SDK
+//
+// All public symbols in this file are intended to allow interoperation with Kotlin Multiplatform for other PubNub frameworks.
+// While these symbols are public, they are intended strictly for internal usage.
+//
+// External developers should refrain from directly using these symbols in their code, as their implementation details
+// may change in future versions of the framework, potentially leading to breaking changes.
+
+import Foundation
+
+@objc
+public class KMPPresenceChange: NSObject {
+  @objc public let event: String?
+  @objc public let uuid: String?
+  @objc public let occupancy: NSNumber?
+  @objc public let state: KMPAnyJSON?
+  @objc public let channel: String?
+  @objc public let subscription: String?
+  @objc public let timetoken: NSNumber?
+  @objc public let join: [String]?
+  @objc public let leave: [String]?
+  @objc public let timeout: [String]?
+  @objc public let refreshHereNow: NSNumber?
+  @objc public let userMetadata: KMPAnyJSON?
+
+  static func from(change: PubNubPresenceChange) -> [KMPPresenceChange] {
+    change.actions.map { KMPPresenceChange(change: change, action: $0) }
+  }
+
+  private init(change: PubNubPresenceChange, action: PubNubPresenceChangeAction) {
+    occupancy = NSNumber(value: change.occupancy)
+    channel = change.channel
+    subscription = change.subscription
+    timetoken = NSNumber(value: change.timetoken)
+    refreshHereNow = NSNumber(value: change.refreshHereNow)
+    userMetadata = if let value = change.metadata?.rawValue { KMPAnyJSON(value) } else { nil }
+
+    switch action {
+    case .join(let uuids):
+      event = "join"
+      join = uuids
+      uuid = nil
+      state = nil
+      leave = nil
+      timeout = nil
+    case .leave(let uuids):
+      event = "leave"
+      leave = uuids
+      uuid = nil
+      state = nil
+      join = nil
+      timeout = nil
+    case let .stateChange(affectedUUID, newState):
+      event = "state-change"
+      state = KMPAnyJSON(newState.codableValue)
+      uuid = affectedUUID
+      join = nil
+      leave = nil
+      timeout = nil
+    case .timeout(let uuids):
+      event = "interval"
+      timeout = uuids
+      uuid = nil
+      state = nil
+      join = nil
+      leave = nil
+    }
```

Comment:
Just noticed how this `action` enum is confusing. For actual `join`, `leave` and `timeout` events there is always only one `uuid` and only for `interval` there is a list of `joined`, `left` and `timeout` uuids. In this code for `timeout` we use `interval` event for some reason.

Reply:
I understand, but why `PubNubPresenceChangeAction` has those cases with list of `uuids`? List of `uuid` will be received from presence only when `interval` event received. All other `join`, `leave` and `timeout` is for the single `uuid` and I would expect that for those actions we will assign value to the `uuid` field of `KMPPresenceChange`.

Reply:
It was implemented a long time ago before I ever joined. Here's how it works if you get the `uuid` only:

https://github.com/pubnub/swift/blob/master/Sources/PubNub/Models/PubNubPresence.swift#L269-L278

And here's how it's processed if you get the `interval`:

https://github.com/pubnub/swift/blob/master/Sources/PubNub/Models/PubNubPresence.swift#L285-L293

All in all, it's mapped to the associated enum types you mentioned. It's useful from the customer's point of view, they would like to know who joined/left/etc without knowing that we have to handle two possible fields under the hood. It's hidden for them by us.

Reply:
Ok, I see that Swift never provided `uuid` for regular event (non `interval`).